tests/test_runner.py

```diff
@@ -0,0 +1,77 @@
+"""
+Test cases for runner module.
+"""
+# Copyright (c) 2018 Red Hat, Inc. All rights reserved. This copyrighted
+# material is made available to anyone wishing to use, modify, copy, or
+# redistribute it subject to the terms and conditions of the GNU General Public
+# License v.2 or later.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+import os
+import unittest
+
+import mock
+
+from skt import runner
+
+SCRIPT_PATH = os.path.dirname(__file__)
+
+
+class TestRunner(unittest.TestCase):
+    """Test cases for runner module"""
+
+    def test_getrunner(self):
+        """Ensure getrunner() can create a runner subclass"""
+        result = runner.getrunner('beaker', {'jobtemplate': 'test'})
+        self.assertIsInstance(result, runner.beakerrunner)
+
+    def test_invalid_getrunner(self):
+        """Ensure getrunner() throws an exception for an invalid runner"""
+        bad_runner = 'pizza'
+        with self.assertRaises(Exception) as context:
+            runner.getrunner(bad_runner, {'jobtemplate': 'test'})
+            self.assertTrue(
+                "Unknown runner type: {}".format(bad_runner)
+                in context.exception
+            )
+
+    def test_getxml(self):
+        """Ensure beakerrunner.getxml() returns xml"""
+        args = {
+            'jobtemplate': '{}/assets/test.xml'.format(SCRIPT_PATH)
+        }
+        myrunner = runner.beakerrunner(**args)
+        result = myrunner.getxml({'replace_this_text': 'with_this_text'})
+        self.assertTrue('<xml>' in result)
+        self.assertTrue('This is test xml!' in result)
+
+    def test_getxml_replace(self):
+        """Ensure beakerrunner.getxml() returns xml with replacements"""
+        args = {
+            'jobtemplate': '{}/assets/test.xml'.format(SCRIPT_PATH)
+        }
+        myrunner = runner.beakerrunner(**args)
+        result = myrunner.getxml({'TEST': 'replaced text'})
+        self.assertTrue('<xml>' in result)
+        self.assertTrue('This is test xml!' in result)
```

Comment:
You don't seem to test that replacement was made here. Do you think you could check that the whole resulting text matches the expectations?

Reply:
Good catch. Fixed.

Reply:
I mean, could you check that the *whole* returned text matches the expectations? I.e. match against the whole returned text, so that we catch any possible text mangling besides the replacement itself, instead of testing just the addition of "replaced text", and the two sub-strings being intact, which you do now.